{
    "schema_version": "1.6.0",
    "id": "OESA-2024-2141",
    "modified": "2024-09-14T00:00:00Z",
    "published": "2024-09-14T00:00:00Z",
    "withdrawn": "",
    "aliases": null,
    "related": [
        "CVE-2024-0450"
    ],
    "summary": "python3 security update",
    "details": "Python combines remarkable power with very clear syntax. It has modules, classes, exceptions, very high level dynamic data types, and dynamic typing. There are interfaces to many system calls and libraries, as well as to various windowing systems. New built-in modules are easily written in C or C++ (or other languages, depending on the chosen implementation). Python is also usable as an extension language for applications written in other languages that need easy-to-use scripting or automation interfaces.\r\n\r\nSecurity Fix(es):\r\n\r\nAn issue was found in the CPython `zipfile` module affecting versions 3.12.1, 3.11.7, 3.10.13, 3.9.18, and 3.8.18 and prior.\r\n\r\nThe zipfile module is vulnerable to “quoted-overlap” zip-bombs which exploit the zip format to create a zip-bomb with a high compression ratio. The fixed versions of CPython makes the zipfile module reject zip archives which overlap entries in the archive.\r\n\r\n(CVE-2024-0450)",
    "severity": null,
    "affected": [
        {
            "package": {
                "ecosystem": "openEuler-22.03-LTS-SP1",
                "name": "python3",
                "purl": "pkg:rpm/python3@3.9.9-31.oe2203sp1"
            },
            "severity": null,
            "ranges": [
                {
                    "type": "ECOSYSTEM",
                    "repo": "",
                    "events": [
                        {
                            "introduced": "0",
                            "fixed": "3.9.9-31.oe2203sp1",
                            "last_affected": "",
                            "limit": ""
                        }
                    ],
                    "database_specific": null
                }
            ],
            "versions": null,
            "ecosystem_specific": {
                "aarch64": [
                    "python3-3.9.9-31.oe2203sp1.aarch64.rpm",
                    "python3-debug-3.9.9-31.oe2203sp1.aarch64.rpm",
                    "python3-debuginfo-3.9.9-31.oe2203sp1.aarch64.rpm",
                    "python3-debugsource-3.9.9-31.oe2203sp1.aarch64.rpm",
                    "python3-devel-3.9.9-31.oe2203sp1.aarch64.rpm",
                    "python3-unversioned-command-3.9.9-31.oe2203sp1.aarch64.rpm"
                ],
                "noarch": [
                    "python3-help-3.9.9-31.oe2203sp1.noarch.rpm"
                ],
                "src": [
                    "python3-3.9.9-31.oe2203sp1.src.rpm"
                ],
                "x86_64": [
                    "python3-3.9.9-31.oe2203sp1.x86_64.rpm",
                    "python3-debug-3.9.9-31.oe2203sp1.x86_64.rpm",
                    "python3-debuginfo-3.9.9-31.oe2203sp1.x86_64.rpm",
                    "python3-debugsource-3.9.9-31.oe2203sp1.x86_64.rpm",
                    "python3-devel-3.9.9-31.oe2203sp1.x86_64.rpm",
                    "python3-unversioned-command-3.9.9-31.oe2203sp1.x86_64.rpm"
                ]
            },
            "database_specific": null
        }
    ],
    "references": [
        {
            "type": "ADVISORY",
            "url": "https://nvd.nist.gov/vuln/detail/CVE-2024-0450"
        }
    ],
    "credits": null,
    "database_specific": {
        "severity": "Medium"
    }
}